api/src/test/java/uk/gov/dhsc/htbhf/claimant/entitlement/CycleVoucherEntitlementTest.java

```diff
@@ -0,0 +1,76 @@
+
+        assertThat(thrown.getMessage()).isEqualTo("List of voucher entitlements must not be null or empty.");
+    }
+
+    @Test
+    void shouldThrowExceptionWhenListIsEmpty() {
+        IllegalArgumentException thrown = catchThrowableOfType(() -> new CycleVoucherEntitlement(emptyList()), IllegalArgumentException.class);
+
+        assertThat(thrown.getMessage()).isEqualTo("List of voucher entitlements must not be null or empty.");
+    }
+
+    @Test
+    void shouldCreateTotalValuesFromListOfVoucherEntitlements() {
+        VoucherEntitlement voucherEntitlement1 = VoucherEntitlement.builder()
+                .vouchersForChildrenUnderOne(1)
+                .vouchersForChildrenBetweenOneAndFour(2)
+                .vouchersForPregnancy(3)
+                .voucherValueInPence(100)
+                .build();
+        VoucherEntitlement voucherEntitlement2 = VoucherEntitlement.builder()
+                .vouchersForChildrenUnderOne(2)
+                .vouchersForChildrenBetweenOneAndFour(3)
+                .vouchersForPregnancy(3)
+                .voucherValueInPence(100)
+                .build();
+
+        CycleVoucherEntitlement result = new CycleVoucherEntitlement(asList(voucherEntitlement1, voucherEntitlement2));
+
+        assertThat(result.getVouchersForChildrenUnderOne()).isEqualTo(3);
+        assertThat(result.getVouchersForChildrenBetweenOneAndFour()).isEqualTo(5);
+        assertThat(result.getVouchersForPregnancy()).isEqualTo(6);
+        assertThat(result.getVoucherValueInPence()).isEqualTo(100);
+        assertThat(result.getTotalVoucherEntitlement())
+                .isEqualTo(voucherEntitlement1.getTotalVoucherEntitlement() + voucherEntitlement2.getTotalVoucherEntitlement());
+        assertThat(result.getTotalVoucherValueInPence())
+                .isEqualTo(voucherEntitlement1.getTotalVoucherValueInPence() + voucherEntitlement2.getTotalVoucherValueInPence());
+    }
+
+    @Test
+    void shouldCreateTotalValuesFromListOfVoucherEntitlementWithMissingValues() {
+        VoucherEntitlement voucherEntitlement1 = VoucherEntitlement.builder()
+                .vouchersForChildrenUnderOne(1)
+                .voucherValueInPence(100)
+                .build();
+        VoucherEntitlement voucherEntitlement2 = VoucherEntitlement.builder()
+                .vouchersForChildrenBetweenOneAndFour(1)
+                .voucherValueInPence(100)
+                .build();
+
+        CycleVoucherEntitlement result = new CycleVoucherEntitlement(asList(voucherEntitlement1, voucherEntitlement2));
+
+        assertThat(result.getVouchersForChildrenUnderOne()).isEqualTo(1);
+        assertThat(result.getVouchersForChildrenBetweenOneAndFour()).isEqualTo(1);
+        assertThat(result.getVouchersForPregnancy()).isEqualTo(0);
+        assertThat(result.getVoucherValueInPence()).isEqualTo(100);
+        assertThat(result.getTotalVoucherEntitlement())
```

Comment:
Same here.

Reply:
I didn't want to do the values here as they are calculated internally in the VoucherEntitlement constructor.